examples/pxScene2d/src/pxObject.h

```diff
@@ -595,10 +607,23 @@ class pxObject: public rtObject
   bool mClip;
   bool mMask;
   bool mDraw;
+
+  rtObjectRef             mEffectRef;         // Shader Config objects
+  rtObjectRef             mEffectConfigRef;   // rtMap   is Shader Config object
+  rtObjectRef             mEffectArrayRef;    // rtArray of Shader Config objects
+
+  rtObjectRef             mEffects;           // rtArray of Shader Config objects
+  rtShaderResource       *mEffectShaderPtr;   // Shader Effect used by this pxObject
```

Comment:
Should this be a rtShaderResourceRef?  Is there a reason this is stored as a raw pointer?

Reply:
Refactored